src/txn/txn_rollback_to_stable.c

```diff
@@ -646,32 +646,54 @@ __rollback_abort_ondisk_kv(WT_SESSION_IMPL *session, WT_REF *ref, WT_COL *cip, W
          */
         WT_RET(__wt_page_cell_data_ref(session, page, vpack, &buf));
 
-        WT_ERR(__wt_upd_alloc(session, &buf, WT_UPDATE_STANDARD, &upd, NULL));
         /*
-         * Set the transaction id of updates to WT_TXN_NONE when called from recovery, because the
-         * connections write generation will be initialized after rollback to stable and the updates
-         * in the cache will be problematic. The transaction id of pages which are in disk will be
-         * automatically reset as part of unpacking cell when loaded to cache.
+         * For prepared transactions, it is possible that both the on-disk key start and stop time
```

Comment:
Why we have to check vpack->tw.durable_stop_ts == WT_TS_NONE in vpack->tw.durable_stop_ts == WT_TS_NONE &&
           __rollback_check_if_txnid_non_committed(session, vpack->tw.stop_txn)

Reply:
That was added as part of a standalone release aborting the updates from a transaction where there are no timestamps involved. It is just an additional check just to make sure that it didn't break anything for MongoDB. As of now, we don't see any problems, so that check can be removed.